src/coreclr/jit/importer.cpp

```diff
@@ -7827,8 +7827,11 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                     op1->gtFlags |= GTF_RELOP_NAN_UN | GTF_UNSIGNED;
                 }
 
-                // See if we can optimize type comparisons
-                op1 = gtFoldTypeCompare(op1);
+                // See if we can optimize type comparisons, IsCall() checks for better TP
+                if (op1->gtGetOp1()->IsCall() && op1->gtGetOp2()->IsCall())
+                {
+                    op1 = gtFoldTypeCompare(op1);
+                }
```

Comment:
Should this have a `compDbgCode` check? Doesn't it otherwise give weird results when stepping through in debug?

Reply:
actually going to remove this, it turns out it only optimizes hand written IL like `ldtoken ldtoken ceq` while normally `typeof() == typeof()` is optimized in `NI_System_Type_op_Equality` intrinsic expansion (which is expanded only in non-debug tier0)